core/src/main/java/com/linecorp/armeria/server/VerbSuffixPathMapping.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+
+final class VerbSuffixPathMapping extends AbstractPathMapping {
+    private static final Pattern VERB_PATTERN = Pattern.compile("[^/]+:([a-zA-Z0-9-_]+)$");
```

Comment:
Q) Is there a reference why these characters were chosen? As per [HttpRule](https://cloud.google.com/endpoints/docs/grpc-service-config/reference/rpc/google.api#google.api.HttpRule), `Verb` is represented by `LITERAL`.
> The syntax LITERAL matches literal text in the URL path. If the LITERAL contains any reserved character, such characters should be percent-encoded before the matching.

What do you think of adding `%.~` as allowed patterns? https://datatracker.ietf.org/doc/html/rfc3986#section-2.3

Reply:
I don't actually have no idea about it as it was implemented when I took this PR. I think we should add `%.~` in the regex according to the RFC. Thank you for your suggestion!
I will add some tests about the verb pattern.